bbc/simorgh
# Transcript First Video - BBC One Hackathon Dec 23

One BBC December 2023 Hackathon  - Transcript First Video team
(Winner of Panel awards)

Front end subteam Isabella Mitchell, Alberto Lazzarin, Indu Aggarwal

Overall changes
======
Renders a transcript for media.
Note, during the hackathon we deployed to the preview pipeline so we could use real assets. To see the transcript you can follow the local testing steps below.


https://github.com/bbc/simorgh/assets/98817636/8e89aaa9-17f8-45b3-9eb0-80c15f45886e


https://github.com/bbc/simorgh/assets/98817636/8c232e70-0be9-455e-9c43-32fb44b60bc9



Code changes
======
- Adds transcript component
- Adds dummy transcript data (auto transcribed & translated using Eurovox)
- Artificially inserts transcript data into relevant articles. (In production this step would be handled by backend logic).


Testing
======
1. Pull down this PR
2. Use `yarn` and `yarn dev` to run locally
3. Visit http://localhost:7080/mundo/articles/c4ym969m7w1o?renderer_env=test to see the transcript component on a Media Article page.
4. Visit http://localhost:7080/mundo/articles/cle16n19nd9o?renderer_env=test to see the transcript component on an Article page.

(Note, renderer_env=test gets data from the BFF. You may require certificates for this).

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
